pkgs/by-name/du/dune3d/package.nix

```diff
@@ -10,33 +11,48 @@
   librsvg,
   libspnav,
   libuuid,
+  libxml2,
   meson,
   ninja,
-  opencascade-occt,
+  opencascade-occt_7_6,
   pkg-config,
   python3,
   stdenv,
   wrapGAppsHook3,
 }:
 
-stdenv.mkDerivation rec {
+let
+  opencascade-occt = opencascade-occt_7_6;
+in
+stdenv.mkDerivation (finalAttrs: {
   pname = "dune3d";
   version = "1.1.0";
 
   src = fetchFromGitHub {
     owner = "dune3d";
     repo = "dune3d";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-Z/kdOc/MbnnEyRsel3aZGndTAy1eCdAK0Wdta0HxaE4=";
   };
 
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    # uuid.h is bundled with system frameworks on darwin
+    substituteInPlace meson.build \
+      --replace-fail "uuid = dependency('uuid')" "" \
+      --replace-fail "build_dependencies += uuid" ""
+  '';
+
   nativeBuildInputs = [
     gobject-introspection
     meson
     ninja
     pkg-config
     wrapGAppsHook3
+    libxml2 # for xmllint
+  ] ++ lib.optionals stdenv.isDarwin [
+    clang_17 # required by use of std::format
```

Comment:
Change to whole stdenv to clang if the package is built on Darwin. (cf. the project's flake.nix)